geoscilabs/gravity/gravityDike.py

```diff
@@ -0,0 +1,328 @@
+    axeToDraw.set_ylabel("Y (m)", fontsize=16)
+    Step = pow(stationSpacing, 0.5)
+    scalax, scalay, color = graphaDataGenerator(
+        delta_rho, z1, z2, b, beta, maxR, maxScala, Step
+    )
+    dat0 = axeToDraw.scatter(
+        scalax, scalay, c=color, cmap="plasma", marker="s", s=450 * pow(Step, 0.5)
+    )
+    axeToDraw.set_title(r"$\Delta g_z$" + "(mGal)", fontsize=16)
+    plt.colorbar(dat0, ax=axeToDraw)
+
+
+# get the data of the third picture
+def graphaDataGenerator(delta_rho, z1, z2, b, beta, maxR, maxScala, Step):
+    scalax = []
+    scalay = []
+    color = []
+    for i in np.arange(-maxR, maxR + Step, Step):
+        g = calculategravity(delta_rho, z1, z2, b, beta, i)
+        for j in np.arange(-maxR, maxR + Step, Step):
+            scalax.append(i)
+            scalay.append(j)
+            color.append(g)
+    return scalax, scalay, color
+
+
+# draw the widgets
+def interact_gravity_Dike():
+    s1 = FloatSlider(
+        description=r"$\Delta\rho$",
+        min=-5.0,
+        max=5.0,
+        step=0.1,
+        value=1.0,
+        continuous_update=False,
+    )
+    s2 = FloatSlider(
+        description=r"$z_1$",
+        min=0.1,
+        max=4.0,
+        step=0.1,
+        value=1 / 3,
+        continuous_update=False,
+    )
+    s3 = FloatSlider(
+        description=r"$z_2$",
+        min=0.1,
+        max=5.0,
+        step=0.1,
+        value=4 / 3,
+        continuous_update=False,
+    )
+    s4 = FloatSlider(
+        description="b", min=0.1, max=5.0, step=0.1, value=1.0, continuous_update=False
+    )
+    s5 = FloatSlider(
+        description=r"$\beta$",
+        min=-85,
+        max=85,
+        step=5,
+        value=45,
+        continuous_update=False,
+    )
+    s6 = FloatSlider(
+        description="Step",
+        min=0.005,
+        max=0.10,
+        step=0.005,
+        value=0.01,
+        continuous_update=False,
+        readout_format=".3f",
+    )
+    b1 = ToggleButton(
+        value=True,
+        description="keep previous plots",
+        disabled=False,
+        button_style="",  # 'success', 'info', 'warning', 'danger' or ''
+        tooltip="Click me",
+        layout=Layout(width="20%"),
+    )
+    v1 = VBox([s1, s2, s3])
+    v2 = VBox([s4, s5, s6])
+    out1 = HBox([v1, v2, b1])
+    out = interactive_output(
+        drawfunction,
+        {
+            "delta_rho": s1,
+            "z1": s2,
+            "z2": s3,
+            "b": s4,
+            "beta": s5,
+            "stationSpacing": s6,
+            "B": b1,
+        },
+    )
+    return VBox([out1, out])
+
+
+# Print the result of the last data
+def printResult():
+    global currentResult
+    print("{0: ^10}{1: ^10}".format("X", "Δgz"))
+    for i in range(len(currentResult[0])):
+        print(
+            "{0: ^10}{1: ^10}".format(
+                "%.3f" % currentResult[0][i], "%.6f" % currentResult[1][i]
+            )
+        )
+
+
+if __name__ == "__main__":

```

Comment:
this doesn't need to be here. It is really only necessary if you expect people to try and run this file as a python script, but users will be interacting with it by importing the package.